File: Documentation/v20.1/Content/Samples/Samplescsharp/AcDbMgdExtensions/DatabaseServices/DatabaseExtensionsCommands.cs
using Autodesk.AutoCAD.DatabaseServices;
using Autodesk.AutoCAD.EditorInput;
using Autodesk.AutoCAD.Runtime;
using Autodesk.AutoCAD.ApplicationServices;

[assembly: CommandClass(typeof(Samplescsharp.AcDbMgdExtensions.DatabaseServices.DatabaseExtensionsCommands))]
namespace Samplescsharp.AcDbMgdExtensions.DatabaseServices
{
   
    public class DatabaseExtensionsCommands : CommandClass
    {

        #region BlockTable
        [CommandMethod("BlockTable")]
        public void BlockTable()
        {
            using (Transaction trx = Db.TransactionManager.StartTransaction())
            {
                BlockTable bt = Db.BlockTable();
                Ed.WriteLine(bt.ObjectId);
            }
        }
        #endregion

        #region BlockTableTrx
        [CommandMethod("BlockTableTrx")]
        public void BlockTableTrx()
        {
            using (Transaction trx = Db.TransactionManager.StartTransaction())
            {
                BlockTable bt = Db.BlockTable(trx);
                Ed.WriteLine(bt.ObjectId);
            }
        }
        #endregion


        #region DimStyleTable
        [CommandMethod("DimStyleTable")]
        public void DimStyleTable()
        {
            using (Transaction trx = Db.TransactionManager.StartTransaction())
            {
                DimStyleTable dt = Db.DimStyleTable();
                Ed.WriteLine(dt.ObjectId);
            }
        }
        #endregion

        #region DimStyleTableTrx
        [CommandMethod("DimStyleTableTrx")]
        public void DimStyleTableTrx()
        {
            using (Transaction trx = Db.TransactionManager.StartTransaction())
            {
                DimStyleTable dt = Db.DimStyleTable(trx);
                Ed.WriteLine(dt.ObjectId);
            }
        }
        #endregion


        #region LayerTable
        [CommandMethod("LayerTable")]
        public void LayerTable()
        {
            using (Transaction trx = Db.TransactionManager.StartTransaction())
            {
                LayerTable dt = Db.LayerTable();
                Ed.WriteLine(dt.ObjectId);
            }
        }
        #endregion

        #region LayerTableTrx
        [CommandMethod("LayerTableTrx")]
        public void LayerTableTrx()
        {
            using (Transaction trx = Db.TransactionManager.StartTransaction())
            {
                LayerTable lt = Db.LayerTable(trx);
                Ed.WriteLine(lt.ObjectId);
            }
        }
        #endregion
    }
}

File: Documentation/v20.1/Content/Samples/Samplesvb/AcDbMgdExtensions/DatabaseServices/DatabaseExtensionsCommands.vb
Imports Autodesk.AutoCAD.DatabaseServices
Imports Autodesk.AutoCAD.EditorInput
Imports Autodesk.AutoCAD.Runtime
<Assembly: CommandClass(GetType(Samplesvb.AcDbMgdExtensions.DatabaseServices.DatabaseExtensionsCommands))>

Namespace AcDbMgdExtensions.DatabaseServices

    Public Class DatabaseExtensionsCommands : Inherits CommandClass

        '#Region "BlockTable"
        <CommandMethod("BlockTable")>
        Public Sub BlockTable()
            Using trx As Transaction = Db.TransactionManager.StartTransaction
                Dim bt As BlockTable = Db.BlockTable
                Ed.WriteLine(bt.ObjectId)
            End Using
        End Sub
        '#End Region


        '#Region "BlockTableTrx"
        <CommandMethod("BlockTableTrx")>
        Public Sub BlockTableTrx()
            Using trx As Transaction = Db.TransactionManager.StartTransaction
                Dim bt As BlockTable = Db.BlockTable(trx)
                Ed.WriteLine(bt.ObjectId)
            End Using
        End Sub
        '#End Region


        '#Region "DimStyleTable"
        <CommandMethod("DimStyleTable")>
        Public Sub DimStyleTable()
            Using trx As Transaction = Db.TransactionManager.StartTransaction
                Dim dt As DimStyleTable = Db.DimStyleTable
                Ed.WriteLine(dt.ObjectId)
            End Using
        End Sub
        '#End Region


        '#Region "DimStyleTableTrx"
        <CommandMethod("DimStyleTableTrx")>
        Public Sub DimStyleTableTrx()
            Using trx As Transaction = Db.TransactionManager.StartTransaction
                Dim dt As DimStyleTable = Db.DimStyleTable(trx)
                Ed.WriteLine(dt.ObjectId)
            End Using
        End Sub
        '#End Region


             '#Region "LayerTable"
        <CommandMethod("LayerTable")>
        Public Sub LayerTable()
            Using trx As Transaction = Db.TransactionManager.StartTransaction
                Dim lt As LayerTable = Db.LayerTable
                Ed.WriteLine(lt.ObjectId)
            End Using
        End Sub
        '#End Region


        '#Region "LayerTableTrx"
        <CommandMethod("LayerTableTrx")>
        Public Sub LayerTableTrx()
            Using trx As Transaction = Db.TransactionManager.StartTransaction
                Dim lt As LayerTable = Db.LayerTable(trx)
                Ed.WriteLine(lt.ObjectId)
            End Using
        End Sub
        '#End Region






    End Class
End Namespace